### PR TITLE
Updated docs: Execution Provider overview

### PR DESCRIPTION
**Pull request by manashgoswami**
**Description**: Intro to EPs and overview description

**Motivation and Context**
- to explain the EP concept, built EPs, API and usage

### CODE REVIEW

**Comment by pranavsharma** [comment]
All usage examples in the specific EP pages are wrong. See [this](https://github.com/microsoft/onnxruntime/blob/master/docs/execution_providers/TensorRT-ExecutionProvider.md#cc) as an example. It uses the internal C++ API. That's not how users use our EPs. We should fix this. Doesn't have to be done in this PR.

File: docs/execution_providers/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+ONNX Runtime supports many different execution providers today. Some of the EPs are in GA and used in live service. Many are in released in preview to enable developers to develop and customize their application using the different options.
+
+### Adding an Execution Provider
+
+Developers of specialized HW acceleration solutions can integrate with ONNX Runtime to execute ONNX models on their stack. To create an EP to interface with ONNX Runtime you must first identify a unique name for the EP. Follow the steps outlined [here](../AddingExecutionProvider.md) to integrate your code in the repo.
+
+### Building ONNX Runtime package with EPs
+
+The ONNX Runtime package can be built with any combination of the EPs along with the default CPU execution provider. **Note** that if multiple EPs are combined into the same ONNX Runtime package then all the dependent librariers must be present in the execution environment. The steps for producing the ONNX Runtime package with different EPs is documented [here](../../BUILD.md#execution-providers).
```
Comment (faxu) [inline docs/execution_providers/README.md:19]: typo

File: docs/execution_providers/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+### APIs for Execution Provider
+
+The same ONNX Runtime API is used across all EPs. This provides the consistent interface for applications to run with different HW acceleration platforms. The APIs to set EP options are available across Python, C/C++, Java and node.js. **Note** we are updating our API support to get parity across all language binding and will update specifics here.
```
Comment (faxu) [inline docs/execution_providers/README.md:23]: C#?

File: docs/execution_providers/README.md (hunk 1 above)
Comment (manashgoswami) [reply docs/execution_providers/README.md]: updated.

File: docs/execution_providers/README.md (hunk 2 above)
Comment (manashgoswami) [reply docs/execution_providers/README.md]: added

**Comment by jywu-msft** [comment]
> All usage examples in the specific EP pages are wrong. See [this](https://github.com/microsoft/onnxruntime/blob/master/docs/execution_providers/TensorRT-ExecutionProvider.md#cc) as an example. It uses the internal C++ API. That's not how users use our EPs. We should fix this. Doesn't have to be done in this PR.

yes. @oliviajain will help take a look at that.

File: docs/execution_providers/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import onnxruntime as rt
+
+# initialize the model.onnx
+sess = rt.InferenceSession("model.onnx")
+
+#define the priority order for the execution providers
+EP_list = ['CUDAExecutionProvider', 'CPUExecutionProvider']
+
+# set the EP priority
+sess.set_providers(EP_list)
```
Comment (jywu-msft) [inline docs/execution_providers/README.md:43]: I think the a better example would be to create session with explicit providers list.
and then later changing providers using set_providers api.

```
# prefer CUDA Execution Provider over CPU Execution Provider
EP_list = ['CUDAExecutionProvider', 'CPUExecutionProvider']
sess = rt.InferenceSession("model.onnx", providers=EP_list)
sess.run(...)
# force execution on CPU
EP_list_updated = ['CPUExecutionProvider']
sess.set_providers(EP_list_updated)
sess.run(...)
```

Comment (manashgoswami) [reply docs/execution_providers/README.md]: updated sample.

File: docs/execution_providers/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# Introduction
+
+ONNX Runtime is capable of working with different HW acceleration libraries to execute the ONNX models on the hardware platform. ONNX Runtime supports an extensible framework, called **Execution Providers** (EP), to integrate with the HW specific libraries. This interface enables flexibility for the AP application developer to deploy their ONNX models in different environments in the cloud and the edge and optimize the execution by taking advantage of the compute capabilities of the platform.
```
Comment (natke) [inline docs/execution_providers/README.md:3]: What is AP in this context?

File: docs/execution_providers/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+<p align="center"><img width="50%" src="images/ONNX_Runtime_EP1.png" alt="Executing ONNX models across different HW environments"/></p>
+
+ONNX Runtime works with the execution provider(s) using the `GetCapability()` interface to allocate specific nodes or sub-graphs for execution by the EP library in supported hardware. The EP libraries that are preinstalled in the execution environment processes and executes the ONNX sub-graph on the hardware. This architecture abstracts out the details of the hardware specific libraries that are essential to optimizing the execution of deep neural networks across hardware platforms like CPU, GPU, FPGA or specialized NPUs.
+
+<p align="center"><img width="50%" src="images/ONNX_Runtime_EP3.png" alt="ONNX Runtime GetCapability()"/></p>
+
+ONNX Runtime supports many different execution providers today. Some of the EPs are in GA and used in live service. Many are in released in preview to enable developers to develop and customize their application using the different options.
+
+### Adding an Execution Provider
```
Comment (natke) [inline docs/execution_providers/README.md:13]: This is stylistic: suggest change to "Add an Execution Provider"

File: docs/execution_providers/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Developers of specialized HW acceleration solutions can integrate with ONNX Runtime to execute ONNX models on their stack. To create an EP to interface with ONNX Runtime you must first identify a unique name for the EP. Follow the steps outlined [here](../AddingExecutionProvider.md) to integrate your code in the repo.
+
+### Building ONNX Runtime package with EPs
```
Comment (natke) [inline docs/execution_providers/README.md:17]: Stylistic: Building -> Build

**Review by natke — APPROVED** [review]

**Pull request merged**